### PR TITLE
fix(fab): allow reactive switching between basic and extended style

### PR DESCRIPTION
**Pull request by tychenjiajun**
@vinaypai Check this out.

### CODE REVIEW

File: components/fab/README.md [hunk 1]
```diff
@@ -2,18 +2,28 @@
...
+
+#### Extended FAB
+
 ```html
-<m-fab aria-label="Favorite">
-    <m-fab-icon class="material-icons">favorite</m-fab-icon>
+<m-fab aria-label="Create">
+    <m-icon icon="add" slot="icon"></m-icon>
+    Create
 </m-fab>
```
Comment (vinaypai) [inline components/fab/README.md]: Thanks for updating the doc... I figured the `m-fab-icon` and `m-fab-label` was out of date. I still don't think it's possible to create an extended FAB right now, there's an underlying bug.  The example here still shows a regular FAB unless you go into the Vue dev tools and manually set the $data.extended to true. One could fix it by removing `v-if="extended"` from the div wrapping the slot, but I don't know what the intent was there so I don't want to go mucking around right away.

Comment (vinaypai) [reply components/fab/README.md]: Actually... I was wrong, the bug is different than I thought. There just seems to be some issue with reactivity, if you add a label to a previously plain FAB. I will investigate it further and create a bug report with more details once I figure it out.

Comment (matsp) [reply components/fab/README.md]: Any action needed?

Comment (tychenjiajun) [reply components/fab/README.md]: > 
> 
> Actually... I was wrong, the bug is different than I thought. There just seems to be some issue with reactivity, if you add a label to a previously plain FAB. I will investigate it further and create a bug report with more details once I figure it out.

Can confirm the reactive bug https://codepen.io/chankcccc/pen/qBEzKqv

**Pull request merged**